web/app/js/components/util/TapUtils.jsx

```diff
@@ -216,3 +217,34 @@ export const srcDstColumn = (d, resourceType, ResourceLink) => {
     </Popover>
   );
 };
+
+export const tapLink = (d, resourceType, PrefixedLink) => {
+  let namespace = d.sourceLabels.namespace;
+  let resource = "";
+
+  if (_.has(d.sourceLabels, resourceType)) {
+    resource = `${resourceType}/${d.sourceLabels[resourceType]}`;
+  } else if (_.has(d.sourceLabels, "pod")) {
+    resource = `pod/${d.sourceLabels.pod}`;
+  } else {
+    return null; // can't tap a resource by IP from the web UI
+  }
+
+  let toNamespace = "";
+  let toResource = "";
+
+  if (_.has(d.destinationLabels, resourceType)) {
```

Comment:
do we care about doing the same fallback to `pod` here as we do for the sourceLabels? Or is it less important because a tap can be done without the `to` resources?

Reply:
Hmm, that's a good point -- I'll the fallback to pods for consistency.